app/scripts/controllers/transactions/index.js

```diff
@@ -2096,9 +2160,36 @@ export default class TransactionController extends EventEmitter {
     } else if (contractInteractionTypes) {
       transactionType = TRANSACTION_TYPES.CONTRACT_INTERACTION;
       transactionContractMethod = transactions[id]?.contractMethodName;
+      if (
+        transactionContractMethod === 'Approve' &&
```

Comment:
@brad-decker You are right! I will create a constant for these contract method names. It is much better to use constant instead of using the string value directly.

Reply:
@brad-decker Done [here](https://github.com/MetaMask/metamask-extension/pull/15289/commits/546ec4254f645b25569630603b05fe1cb015b9a1).